src/services/DirectoryService/DirectoryService.ts

```diff
@@ -65,9 +68,45 @@ export const getWorkspacePages = (siteName: string): Promise<PageData[]> => {
 export const getMediaData = ({
   siteName,
   mediaDirectoryName,
-}: MediaDirectoryParams): Promise<DirectoryData[]> => {
+  // NOTE: These defaults are set to adhere to our current behaviour.
+  // Github has a fixed upper limit of 1000 items and we return
+  // every item up to the upper limit.
+  // Hence, this behaviour is essentially a single page of up to 1000 items.
```

Comment:
wait sanity check here, out limit is not the github limit per se right? do we not do pagination in our backend code when calling github?

Reply:
Also, from a perf pov, should we use a lower amount (ideally should be = to the number of files typically displayed to a user before any scrolling occurs) to fully utilise the benefits of pagination?

Reply:
> do we not do pagination in our backend code when calling github

github doesn't have pagination for repository contents :/

> should we use a lower amount (ideally should be = to the number of files typically displayed to a user before any scrolling occurs) to fully utilise the benefits of pagination 

the caller handles the limit/page. this default setting is done **to preserve compat with existing hooks**